diff --git a/apple/RNCWebViewImpl.m b/apple/RNCWebViewImpl.m
index 7f5c24d6e2cfb26eddbc513bf17a81c24a7273e2..5b1cb0aed7af363f3d9b759372dcb4989cd659a3 100644
--- a/apple/RNCWebViewImpl.m
+++ b/apple/RNCWebViewImpl.m
@@ -95,9 +95,13 @@ - (BOOL)canPerformAction:(SEL)action
 }
 - (void)buildMenuWithBuilder:(id<UIMenuBuilder>)builder API_AVAILABLE(ios(13.0))  {
     if (@available(iOS 16.0, *)) {
-      if(self.menuItems){
+      if(self.menuItems || self.suppressMenuItems){
         [builder removeMenuForIdentifier:UIMenuLookup];
       }
+      if (self.suppressMenuItems) {
+        [builder removeMenuForIdentifier:UIMenuStandardEdit];
+        [builder removeMenuForIdentifier:UIMenuShare];
+      }
     }
     [super buildMenuWithBuilder:builder];
 }
@@ -893,6 +897,9 @@ -(void)setMenuItems:(NSArray<NSDictionary *> *)menuItems {
 -(void)setSuppressMenuItems:(NSArray<NSString *> *)suppressMenuItems {
     _suppressMenuItems = suppressMenuItems;
     _webView.suppressMenuItems = suppressMenuItems;
+    if (@available(iOS 18.0, *)) {
+        _webView.writingToolsBehavior = suppressMenuItems ? UIWritingToolsBehaviorNone : UIWritingToolsBehaviorDefault;
+    }
 }
 
 #if TARGET_OS_IOS
